FeedStoreChallenge/InMemoryFeedStore.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  File.swift
+//  FeedStoreChallenge
+//
+//  Created by Nadheer on 25/02/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+
+public class InMemoryFeedStore: FeedStore {
+	public init() {}
+	
+	private struct InMemoryFeedCache {
+		let feed: [LocalFeedImage]
+		let timestamp: Date
+	}
+	
+	private var cache: InMemoryFeedCache?
+	private let queue = DispatchQueue(label: "\(InMemoryFeedStore.self)Queue", qos: .userInitiated)
+	
+	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+		queue.async(flags: .barrier) {
+			guard self.cache != nil else { return completion(nil) }
+			self.cache = nil
+			completion(nil)
+		}
+	}
+	
+	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
+		queue.async(flags: .barrier) {
```

Comment:
Since the queue is serial (not concurrent), the barrier flag is not necessary. 

```suggestion
		queue.async {
```